pallets/nfts/src/lib.rs

```diff
@@ -1295,80 +1340,128 @@ pub mod pallet {
 		///
 		/// Weight: `O(1)`
 		#[pallet::call_index(15)]
-		#[pallet::weight(T::WeightInfo::approve_transfer())]
+		#[pallet::weight(T::WeightInfo::approve_transfer(maybe_item.is_some() as u32))]
```

Comment:
Number of items in collection can never exceed u32 due to this. We should probably add some integrity test to ensure this.

Reply:
Not sure why it can't exceed due to the code. It is the u32 cast of the boolean value due to `is_some()` not the item ID.